src/main/webapp/app/code-editor/code-editor-instructor-base-container.component.ts

```diff
@@ -0,0 +1,285 @@
+import { CodeEditorContainer } from 'app/code-editor/code-editor-mode-container.component';
+import { OnDestroy, OnInit } from '@angular/core';
+import { ExerciseType } from 'app/entities/exercise';
+import { Observable, Subscription, throwError } from 'rxjs';
+import { ProgrammingExercise, ProgrammingExerciseParticipationService, ProgrammingExerciseService } from 'app/entities/programming-exercise';
+import {
+    Participation,
+    ParticipationService,
+    ProgrammingExerciseStudentParticipation,
+    SolutionProgrammingExerciseParticipation,
+    TemplateProgrammingExerciseParticipation,
+} from 'app/entities/participation';
+import { ActivatedRoute, Router } from '@angular/router';
+import { CourseExerciseService } from 'app/entities/course';
+import { CodeEditorFileService, CodeEditorSessionService, DomainChange, DomainService, DomainType } from 'app/code-editor/service';
+import { TranslateService } from '@ngx-translate/core';
+import { JhiAlertService } from 'ng-jhipster';
+import { catchError, filter, map, tap } from 'rxjs/operators';
+import { ButtonSize } from 'app/shared/components';
+
+export enum REPOSITORY {
+    ASSIGNMENT = 'ASSIGNMENT',
+    TEMPLATE = 'TEMPLATE',
+    SOLUTION = 'SOLUTION',
+    TEST = 'TEST',
+}
+
+export enum LOADING_STATE {
+    CLEAR = 'CLEAR',
+    INITIALIZING = 'INITIALIZING',
+    FETCHING_FAILED = 'FETCHING_FAILED',
+    CREATING_ASSIGNMENT_REPO = 'CREATING_ASSIGNMENT_REPO',
+    DELETING_ASSIGNMENT_REPO = 'DELETING_ASSIGNMENT_REPO',
+}
+
+export abstract class CodeEditorInstructorBaseContainerComponent extends CodeEditorContainer implements OnInit, OnDestroy {
+    ButtonSize = ButtonSize;
```

Comment:
move this and the IntelliJ version into an instructor folder

Reply:
Moved